spec.md

```diff
@@ -0,0 +1,728 @@
+
+```k
+    syntax EEIMethod ::= "EEI.getAccountBalance"
+ // --------------------------------------------
+    rule <k> EEI.getAccountBalance => BAL ... </k>
+         <id> ACCT </id>
+         <account>
+           <acctID>  ACCT </acctID>
+           <balance> BAL  </balance>
+           ...
+         </account>
+```
+
+#### `EEI.getAccountCode`
+
+Get the code of the given account `ACCT`.
+
+1.  Load the value `ACCT` from `eei.call.id`.
+
+2.  Load and return `eei.accounts[ACCT].code`.
+
+```k
+    syntax EEIMethod ::= "EEI.getAccountCode"
+ // -----------------------------------------
+    rule <k> EEI.getAccountCode => ACCTCODE ... </k>
+         <id> ACCT </id>
+         <accounts>
+           <acctID> ACCT </acctID>
+           <code> ACCTCODE </code>
+           ...
+         </accounts>
+```
+
+#### `EEI.getExternalAccountCode : Int`
+
+Get the code of the given account `ACCT`.
+
+1.  Load and return `eei.accounts[ACCT].code`.
+
+```k
+    syntax EEIMethod ::= "EEI.getExternalAccountCode" Int
+ // -----------------------------------------------------
+    rule <k> EEI.getExternalAccountCode ACCT => ACCTCODE ... </k>
+         <accounts>
+           <acctID> ACCT </acctID>
+           <code> ACCTCODE </code>
+           ...
+         </accounts>
+```
+
+#### `EEI.getAccountStorage : Int`
+
+Returns the value at the given `INDEX` in the current executing accounts storage.
+
+1.  Load `ACCT` from `eei.call.id`.
+
+2.  If `eei.accounts[ACCT].storage[INDEX]` exists:
+
+    i.  then: return `eei.accounts[ACCT].storage[INDEX]`.
+
+    ii. else: return `0`.
+
+```k
+    syntax EEIMethod ::= "EEI.getAccountStorage" Int
+ // ------------------------------------------
+    rule <k> EEI.getAccountStorage INDEX => VALUE ... </k>
+         <id> ACCT </id>
+         <account>
+           <acctID> ACCT </acctID>
+           <storage> ... INDEX |-> VALUE ... </storage>
+           ...
+         </account>
+
+    rule <k> EEI.getAccountStorage INDEX => 0 ... </k>
+         <id> ACCT </id>
+         <account>
+           <acctID> ACCT </acctID>
+           <storage> STORAGE </storage>
+           ...
+         </account>
+      requires notBool INDEX in_keys(STORAGE)
+```
+
+#### `EEI.setAccountStorage : Int Int`
+
+At the given `INDEX` in the executing accounts storage, stores the given `VALUE`.
+
+1.  Load `ACCT` from `eei.call.id`.
+
+2.  Set `eei.accounts[ACCT].storage[INDEX]` to `VALUE`.
+
+```k
+    syntax EEIMethod ::= "EEI.setAccountStorage" Int Int
+ // ----------------------------------------------------
+    rule <k>  EEI.setAccountStorage INDEX VALUE => . ... </k>
+         <id> ACCT </id>
+         <account>
+           <acctID> ACCT </acctID>
+           <storage> STORAGE => STORAGE [ INDEX <- VALUE ] </storage>
+           ...
+         </account>
+```
+
+### Logging
+
+#### `EEI.log : List List`
+
+Logging places a user-specified byte strings (`BS1` and `BS2`) on the blockchain Log for external inspection.
```

Comment:
What is a byte string?

Reply:
I've left it unspecified because I'm not sure we should fix a specific data encoding. I'll change it to `List`, which means list of integers, and can be interpreted as a byte-string as needed.